File: chapter-04/exercise-19-bmi-calculator/BmiCalculator.java
import java.math.BigDecimal;
import java.math.RoundingMode;

class BmiCalculator {
  public static void main(String[] args) {
    var height = getd("What is your height? ");
    var weight = getd("What is your weight? ");

    var bmi = (weight / (height * height)) * 703;

    System.out.printf("Your BMI is %.1f%n", bmi);
    if (bmi < 18.5 || bmi > 25) {
      System.out.println("You are " + (bmi < 18.5 ? "under" : "over") + "weight. You should see your doctor.");
    } else {
      System.out.println("You are within the ideal weight range.");
    }
  }

  private static double getd(String prompt) {
    var line = System.console().readLine(prompt);
    try {
      return Double.parseDouble(line);
    } catch (NumberFormatException e) {
      System.out.println("Enter valid number.");
      return getd(prompt);
    }
  }
}

File: exercise-19-bmi-calculator/bmi_calculator.py
def float_input(prompt):
    try:
        return float(input(prompt))
    except ValueError:
        print("Enter valid number.")
        return float_input(prompt)
        
height = float_input("What is your height (inches)? ")
weight = float_input("What is your weight (pounds)? ")

bmi = (weight / (height * height)) * 703

print(f"Your BMI is {bmi:.1f}")
if 18.5 <= bmi <= 25:
    print("You are within the ideal weight range.")
else:
    print(f"You are {'under' if bmi < 18.5 else 'over'}weight.",
           "You should see your doctor.")

    